ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureBlobDataAccessor.java

```diff
@@ -68,23 +73,19 @@
       AzureMetrics azureMetrics) {
     this.clusterName = clusterName;
     this.azureMetrics = azureMetrics;
-    Configuration storageConfiguration = new Configuration();
-    // Check for proxy
-    if (cloudConfig.vcrProxyHost != null) {
+    this.storageConfiguration = new Configuration();
+    // Check for network proxy
+    proxyOptions = (cloudConfig.vcrProxyHost == null) ? null : new ProxyOptions(ProxyOptions.Type.HTTP,
+        new InetSocketAddress(cloudConfig.vcrProxyHost, cloudConfig.vcrProxyPort));
+    if (proxyOptions != null) {
       logger.info("Using proxy: {}:{}", cloudConfig.vcrProxyHost, cloudConfig.vcrProxyPort);
-      try {
-        // TODO: could add vcrProxyScheme to CloudConfig
-        URL proxyUrl = new URL("http", cloudConfig.vcrProxyHost, cloudConfig.vcrProxyPort, "/");
-        storageConfiguration.put(Configuration.PROPERTY_HTTPS_PROXY, proxyUrl.toString());
-        storageConfiguration.put(Configuration.PROPERTY_HTTP_PROXY, proxyUrl.toString());
-      } catch (MalformedURLException e) {
-        throw new IllegalArgumentException("Bad proxy info");
-      }
     }
+    HttpClient client = new NettyAsyncHttpClientBuilder().proxy(proxyOptions).build();
 
-    // TODO: set retry options depending on whether we are serving live data or replicating
+    // TODO: may want to set different retry options depending on live serving or replication mode
     RequestRetryOptions requestRetryOptions = new RequestRetryOptions();
     storageClient = new BlobServiceClientBuilder().connectionString(azureCloudConfig.azureStorageConnectionString)
+        .httpClient(client)
         .retryOptions(requestRetryOptions)
         .configuration(storageConfiguration)
```

Comment:
`storageConfiguration` appears to now be empty. does it still have to be passed into the builder?

Reply:
I originally was setting something in it and removed it.  Figured I'd keep it as a placeholder in case we need it later.